tests/dualtor/test_ipinip.py

```diff
@@ -0,0 +1,145 @@
+    mg_facts = tor.get_extended_minigraph_facts(tbinfo)
+    return mg_facts["minigraph_ptf_indices"][iface]
+
+
+def build_expected_packet_to_server(encapsulated_packet):
+    """Build packet expected to be received by server from the tunnel packet."""
+    inner_packet = encapsulated_packet[IP].payload[IP].copy()
+    # use dummy mac address that will be ignored in mask
+    inner_packet = Ether(src="aa:bb:cc:dd:ee:ff", dst="aa:bb:cc:dd:ee:ff") / inner_packet
+    exp_pkt = mask.Mask(inner_packet)
+    exp_pkt.set_do_not_care_scapy(Ether, "dst")
+    exp_pkt.set_do_not_care_scapy(Ether, "src")
+    exp_pkt.set_do_not_care_scapy(IP, "tos")
+    exp_pkt.set_do_not_care_scapy(IP, "ttl")
+    exp_pkt.set_do_not_care_scapy(IP, "chksum")
+    return exp_pkt
+
+
+def test_dualtor_ipinip_active_tor(
+    apply_mock_dual_tor_tables,
+    apply_mock_dual_tor_kernel_configs,
+    apply_active_state_to_orchagent,
+    build_encapsulated_packet, test_interface, ptfadapter,
+    tbinfo, tor, tunnel_traffic_monitor
+):
+    encapsulated_packet = build_encapsulated_packet
+    iface, _ = test_interface
+
+    exp_ptf_port_index = get_ptf_server_intf_index(tor, tbinfo, iface)
+    exp_pkt = build_expected_packet_to_server(encapsulated_packet)
+
+    ptfadapter.dataplane.flush()
+    ptf_t1_intf = random.choice(get_t1_ptf_ports(tor, tbinfo))
+    logging.info("send encapsulated packet from ptf t1 interface %s", ptf_t1_intf)
+    testutils.send(ptfadapter, int(ptf_t1_intf.strip("eth")), encapsulated_packet, count=1)
+    _, rec_pkt = testutils.verify_packet_any_port(ptfadapter, exp_pkt, ports=[exp_ptf_port_index])
+    rec_pkt = Ether(rec_pkt)
+    logging.info("received decap packet:\n%s", tunnel_traffic_monitor._dump_show_str(rec_pkt))
+    exp_ttl = encapsulated_packet[IP].payload[IP].ttl - 1
+    exp_tos = encapsulated_packet[IP].payload[IP].tos
+    if rec_pkt[IP].ttl != exp_ttl:
+        pytest.fail("the expected ttl should be %s" % exp_ttl)
+    if rec_pkt[IP].tos != exp_tos:
+        pytest.fail("the expected tos should be %s" % exp_tos)
+
+
+def test_dualtor_ipinip_standby_tor(
+    apply_mock_dual_tor_tables,
+    apply_mock_dual_tor_kernel_configs,
+    apply_standby_state_to_orchagent,
+    build_encapsulated_packet, test_interface, ptfadapter,
+    tbinfo, tor, tunnel_traffic_monitor
+):
+    encapsulated_packet = build_encapsulated_packet
+    iface, _ = test_interface
+
+    exp_ptf_port_index = get_ptf_server_intf_index(tor, tbinfo, iface)
+    exp_pkt = build_expected_packet_to_server(encapsulated_packet)
+
+    ptfadapter.dataplane.flush()
+    ptf_t1_intf = random.choice(get_t1_ptf_ports(tor, tbinfo))
+    logging.info("send encapsulated packet from ptf t1 interface %s", ptf_t1_intf)
+    with tunnel_traffic_monitor(tor, existing=False):
+        testutils.send(ptfadapter, int(ptf_t1_intf.strip("eth")), encapsulated_packet, count=1)
+
+    testutils.verify_no_packet_any(ptfadapter, exp_pkt, ports=[exp_ptf_port_index])
```

Comment:
Same as above, please add a timeout.

Reply:
Same as above, the default timeout for the negative scenario is 0.2 second.